src/tuner/nccl_ofi_tuner.c

```diff
@@ -305,25 +340,30 @@ ncclResult_t nccl_ofi_tuner_get_coll_info(
 		goto exit;
 	}
 
+	float(*table)[NCCL_NUM_PROTOCOLS] = (float(*)[NCCL_NUM_PROTOCOLS])collCostTable;
 	int in_out = -1;
+	int algorithm = NCCL_ALGO_UNDEF;
+	int protocol = NCCL_PROTO_UNDEF;
 	nccl_ofi_tuner_point_t p = {.x = nBytes, .y = nccl_ofi_tuner_ctx->dims.num_ranks};
 
 	/* Check all regions */
 	for (int i = 0; i < nccl_ofi_tuner_ctx->num_regions && in_out < 0; i++) {
-		if (nccl_ofi_tuner_ctx->regions[i].algorithm == NCCL_ALGO_NVLS_TREE && nvlsSupport == 0) {
+		algorithm = nccl_ofi_tuner_ctx->regions[i].algorithm;
+		protocol = nccl_ofi_tuner_ctx->regions[i].protocol;
+		if (table[algorithm][protocol] == NCCL_ALGO_PROTO_IGNORE || algorithm >= numAlgo ||
+		    protocol >= numProto) {
```

Comment:
What might cause the second/third condition to be true? Something else has gone wrong before this for that to happen.

Reply:
those conditions are not going to be true today, because NCCL, from what I have seen, is always setting `numAlgo=NCCL_NUM_ALGORITHMS=6` and `numProto=NCCL_NUM_PROTOCOLS=3`.
But those are valid conditions and do not signal an error. 
`numAlgo` and `numProto` are the dimensions of the table. I think that the API is designed so that NCCL could decide to pass a smaller table to the plugin for some reason. Since we scan the regions in an order that is not the same as incrementing the value of algo/proto, we could be checking a region that ends up not being in the table, and so these conditions make it so we ignore that region.